tracer/build/_build/Build.ExplorationTests.cs

```diff
@@ -0,0 +1,252 @@
+            {
+                GitRepositoryUrl = "https://github.com/domaindrivendev/Swashbuckle.AspNetCore.git",
+                Name = "Swashbuckle.AspNetCore",
+                GitRepositoryTag = "v6.2.3",
+                PathToUnitTestProject = "test/Swashbuckle.AspNetCore.SwaggerGen.Test",
+                SupportedFrameworks = new[] { TargetFramework.NET6_0 }
+            },
+            ExplorationTestName.paket => new ExplorationTestDescription()
+            {
+                GitRepositoryUrl = "https://github.com/fsprojects/Paket.git",
+                Name = "Paket",
+                GitRepositoryTag = "6.2.1",
+                PathToUnitTestProject = "tests/Paket.Tests",
+                TestsToIgnore = new[] { "Loading assembly metadata works" },
+                SupportedFrameworks = new[] { TargetFramework.NET461, TargetFramework.NETCOREAPP3_1 }
+            },
+            ExplorationTestName.ilspy => new ExplorationTestDescription()
+            {
+                GitRepositoryUrl = "https://github.com/icsharpcode/ILSpy.git",
+                Name = "ILSpy",
+                GitRepositoryTag = "6.2.1",
+                PathToUnitTestProject = "ICSharpCode.Decompiler.Tests",
+                TestsToIgnore = new[] { "ICSharpCode.Decompiler.Tests", "UseMc", "_net45", "ImplicitConversions", "ExplicitConversions", "ICSharpCode_Decompiler", "NewtonsoftJson_pcl_debug", "NRefactory_CSharp", "Random_TestCase_1" },
+                SupportedFrameworks = new[] { TargetFramework.NET461, TargetFramework.NETCOREAPP3_1 }
+            },
+            _ => throw new ArgumentOutOfRangeException(nameof(name), name, null)
+        };
+
+        return description;
+    }
+}
+
+partial class Build
+{
+    AbsolutePath ExplorationTestsDirectory => RootDirectory / "exploration-tests";
+
+    [Parameter("Indicates name of exploration test to run. If not specified, will run all tests sequentially.")]
+    readonly ExplorationTestName? ExplorationTestName;
+
+    [Parameter("Indicates whether exploration tests should skip repository cloning. Useful for local development. Default false.")]
+    readonly bool ExplorationSkipClone;
+
+    [Parameter("Indicates whether exploration tests should run on latest repository commit. Useful if you want to update tested repositories to the latest tags. Default false.", 
+               List = false)]
+    readonly bool ExplorationCloneLatest;
+
+    Target RunExplorationTests_Debugger
+        => _ => _
+               .Description("Run exploration tests for debugger.")
+               .After(Clean)
+               .Requires(() => ExplorationTestName)
+               .Executes(() =>
+                {
+                    PrepareMonitoringEnvironment_Debugger();
+                    GitCloneAndRunUnitTests(MonitoringType.Debugger);
+                    RunExplorationTestAssertions_Debugger();
+                })
+        ;
+
+    void PrepareMonitoringEnvironment_Debugger()
```

Comment:
```suggestion
    void PrepareForExplorationTest_Debugger()
```
or
```suggestion
    void SetUpExplorationTest_Debugger()
```


This naming is more consistent with `RunExplorationTestAssertions_Debugger`, and the term "monitoring environment" is not something we currently really use anywhere so it's a bit surprising.

Reply:
If the "namespace" is ExplorationTest, so, in the name of consistency, it should be something like this:
`PrepareExplorationTestEnvironment_Debugger` -> `RunExplorationTestsGitCloneAndUnitTest` -> `RunExplorationTestAssertions_Debugger`

Reply:
Sounds good, just nitpicking here but I would personally prefer `SetUpExplorationTest_Debugger` over `PrepareExplorationTestEnvironment_Debugger` because the term `Environment` makes me think you're talking about Environment Variables, whereas this method will likely need to do other things as well (set up a mock agent, etc etc).

Reply:
Renamed with `SetUpExplorationTest_*`